src/proxyNetworkProvider.ts

```diff
@@ -125,17 +126,32 @@ export class ProxyNetworkProvider implements INetworkProvider {
         }
     }
 
-    async getDefinitionOfFungibleToken(_tokenIdentifier: string): Promise<DefinitionOfFungibleTokenOnNetwork> {
-        // TODO: Implement wrt.:
-        // https://github.com/ElrondNetwork/api.elrond.com/blob/main/src/endpoints/esdt/esdt.service.ts#L221
-        throw new Error("Method not implemented.");
+    async getDefinitionOfFungibleToken(tokenIdentifier: string): Promise<DefinitionOfFungibleTokenOnNetwork> {
+        let encodedTokenIdentifier = Buffer.from(tokenIdentifier).toString("hex");
+
+        let queryResponse = await this.queryContract({
+            address: EsdtContractAddress,
+            func: "getTokenProperties",
+            getEncodedArguments: () => [encodedTokenIdentifier]
+        });
+
+        let dataParts = queryResponse.getReturnDataParts();
+        let definition = DefinitionOfFungibleTokenOnNetwork.fromResponseOfGetTokenProperties(tokenIdentifier, dataParts);
+        return definition;
     }
 
-    async getDefinitionOfTokenCollection(_collection: string): Promise<DefinitionOfTokenCollectionOnNetwork> {
-        // TODO: Implement wrt.:
-        // https://github.com/ElrondNetwork/api.elrond.com/blob/main/src/endpoints/collections/collection.service.ts
-        // https://docs.elrond.com/developers/esdt-tokens/#get-esdt-token-properties
-        throw new Error("Method not implemented.");
+    async getDefinitionOfTokenCollection(collection: string): Promise<DefinitionOfTokenCollectionOnNetwork> {
+        let encodedCollection = Buffer.from(collection).toString("hex");
```

Comment:
besides the `let definition = DefinitionOfFungibleTokenOnNetwork` and `let definition = DefinitionOfTokenCollectionOnNetwork.` lines, the rest of the code is common and can be extracted into a function

Reply:
Fixed.